CONTRIBUTING.md

```diff
@@ -252,6 +252,71 @@ There may, at times, need to be exceptions where breaking changes are allowed in
 discretion of the project's maintainers, and must be carefully considered before merging. An example of an allowed
 breaking change might be a fix for a behavioral bug that was released in an initial minor version (such as `v0.3.0`).
 
+
+## API conventions
+
+In general we adhere to the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required).
+We have a small set of exceptions due to the fact that we're using CRDs while the API conventions predates the introduction 
+of CRDs and the way our controllers only patch actual changes to the status instead of updating the entire status in every 
+reconciliation.
+
+### Optional vs. Required
+
+Status fields must be optional, because our controllers are patching selected fields instead of updating the entire status 
+in every reconciliation.
```

Comment:
Is that true? I thought a patch could patch any set of fields without needing to worry about the others, this is why there are different types of merge patches right? 

I don't know that I've seen this requirement defined anywhere else before, is there some source that lead you to this? Might be worth linking it here for a cross reference

Reply:
@vincepri can you please provide some additional context?

Reply:
> Is that true? I thought a patch could patch any set of fields without needing to worry about the others, this is why there are different types of merge patches right?

This is correct, and that's also why it might fail validation if a field is required. If you have a field is required, and it never gets set during let's say one of the first reconciliation (example: we're waiting for external objects to reconcile first), the api server is going to refuse any patches that doesn't have the said required field

Reply:
If someone else is curious, if a new required field would be added (but not set with patch), the error returned by the patchHelper looks roughly like this:
```bash
Cluster.cluster.x-k8s.io "quick-start-7x5iac" is invalid: [status.requiredBool: Required value, status.requiredBoolPointer: Required value]
```

Reply:
Today I learned! Thanks for the info and trying it out. I wonder if this is documented somewhere that we could link to back up this statement 🤔